test/e2e/suite/trustpolicy/trustpolicy_test.go

```diff
@@ -0,0 +1,467 @@
+			// update trustpolicy.json
+			vhost.SetOption(AddTrustPolicyOption("multiple_registry_scope_trustpolicy.json"))
+
+			// generate an artifact with given repository name
+			artifact2 := GenerateArtifact("", "test-repo2")
+			artifact3 := GenerateArtifact("", "test-repo3")
+
+			// test localhost:5000/test-repo2
+			OldNotation().Exec("sign", artifact2.ReferenceWithDigest()).MatchKeyWords(SignSuccessfully)
+			notation.Exec("verify", artifact2.ReferenceWithDigest()).MatchKeyWords(VerifySuccessfully)
+
+			// test localhost:5000/test-repo3
+			OldNotation().Exec("sign", artifact3.ReferenceWithDigest()).MatchKeyWords(SignSuccessfully)
+			notation.Exec("verify", artifact3.ReferenceWithDigest()).MatchKeyWords(VerifySuccessfully)
+		})
+	})
+
+	It("test registryScope with any(*) repository", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			// generate an artifact with given repository name
+			artifact4 := GenerateArtifact("", "test-repo4")
+			artifact5 := GenerateArtifact("", "test-repo5")
+
+			// test localhost:5000/test-repo4
+			OldNotation().Exec("sign", artifact4.ReferenceWithDigest()).MatchKeyWords(SignSuccessfully)
+			notation.Exec("verify", artifact4.ReferenceWithDigest()).MatchKeyWords(VerifySuccessfully)
+
+			// test localhost:5000/test-repo5
+			OldNotation().Exec("sign", artifact5.ReferenceWithDigest()).MatchKeyWords(SignSuccessfully)
+			notation.Exec("verify", artifact5.ReferenceWithDigest()).MatchKeyWords(VerifySuccessfully)
+		})
+	})
+
+	It("test overlap registryScope", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			// update trustpolicy.json
+			vhost.SetOption(AddTrustPolicyOption("overlapped_registry_scope_trustpolicy.json"))
+
+			artifact := GenerateArtifact("", "test-repo6")
+
+			// test localhost:5000/test-repo
+			OldNotation().Exec("sign", artifact.ReferenceWithDigest()).MatchKeyWords(SignSuccessfully)
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest()).
+				MatchErrKeyWords("registry scope \"localhost:5000/test-repo6\" is present in multiple trust policy statements")
+		})
+	})
+
+	It("test any plus specific repo registryScope", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			// update trustpolicy.json
+			vhost.SetOption(AddTrustPolicyOption("any_plus_other_registry_scope_trustpolicy.json"))
+
+			artifact := GenerateArtifact("", "test-repo7")
+
+			// test localhost:5000/test-repo
+			OldNotation().Exec("sign", artifact.ReferenceWithDigest()).MatchKeyWords(SignSuccessfully)
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest()).
+				MatchErrKeyWords("trust policy statement \"e2e\" uses wildcard registry scope '*', a wildcard scope cannot be used in conjunction with other scope values")
+		})
+	})
+
+	It("test invalid registryScope", func() {
```

Comment:
Do we have an example with a malformed repository format in the registryScope?

Reply:
Added now.